.gitlab-ci.yml

```diff
@@ -1120,6 +1139,166 @@ workflow:
   - when: manual
     allow_failure: true
 
+.on_asc_or_e2e_changes_or_manual:
+  - <<: *if_disable_e2e
+    when: never
+  - <<: *if_auto_e2e
+    when: on_success
+  - <<: *if_main_branch
+    when: on_success
+  - <<: *if_release_branch
+    when: on_success
+  - if: $CI_COMMIT_TAG =~ /^[0-9]+\.[0-9]+\.[0-9]+-rc\.[0-9]+$/
+    when: on_success
+  - <<: *if_mergequeue
+    when: never
+  - changes:
+      paths:
+        # TODO: Add paths that should trigger tests for ASC
+        - test/new-e2e/tests/agent-shared-components/**/*
+  - when: manual
+    allow_failure: true
+
+.on_subcommands_or_e2e_changes_or_manual:
+  - <<: *if_disable_e2e
+    when: never
+  - <<: *if_auto_e2e
+    when: on_success
+  - <<: *if_main_branch
+    when: on_success
+  - <<: *if_release_branch
+    when: on_success
+  - if: $CI_COMMIT_TAG =~ /^[0-9]+\.[0-9]+\.[0-9]+-rc\.[0-9]+$/
+    when: on_success
+  - <<: *if_mergequeue
+    when: never
+  - changes:
+      paths:
+        # TODO: Add paths that should trigger tests for subcommands
+        - test/new-e2e/tests/agent-subcommands/**/*
+  - when: manual
+    allow_failure: true
+
+.on_language-detection_or_e2e_changes_or_manual:
+  - <<: *if_disable_e2e
+    when: never
+  - <<: *if_auto_e2e
+    when: on_success
+  - <<: *if_main_branch
+    when: on_success
+  - <<: *if_release_branch
+    when: on_success
+  - if: $CI_COMMIT_TAG =~ /^[0-9]+\.[0-9]+\.[0-9]+-rc\.[0-9]+$/
+    when: on_success
+  - <<: *if_mergequeue
+    when: never
+  - changes:
+      paths:
+        # TODO: Add paths that should trigger tests for language detection
+        - test/new-e2e/tests/language-detection/**/*
+  - when: manual
+    allow_failure: true
+
+.on_npm_or_e2e_changes_or_manual:
+  - <<: *if_disable_e2e
+    when: never
+  - <<: *if_auto_e2e
+    when: on_success
+  - <<: *if_main_branch
+    when: on_success
+  - <<: *if_release_branch
+    when: on_success
+  - if: $CI_COMMIT_TAG =~ /^[0-9]+\.[0-9]+\.[0-9]+-rc\.[0-9]+$/
+    when: on_success
+  - <<: *if_mergequeue
+    when: never
+  - changes:
+      paths:
+        # TODO: Add paths that should trigger tests for npm
+        - test/new-e2e/tests/npm/**/*
+  - when: manual
+    allow_failure: true
+
+.on_aml_or_e2e_changes_or_manual:
+  - <<: *if_disable_e2e
+    when: never
+  - <<: *if_auto_e2e
+    when: on_success
+  - <<: *if_main_branch
+    when: on_success
+  - <<: *if_release_branch
+    when: on_success
+  - if: $CI_COMMIT_TAG =~ /^[0-9]+\.[0-9]+\.[0-9]+-rc\.[0-9]+$/
+    when: on_success
+  - <<: *if_mergequeue
+    when: never
+  - changes:
+      paths:
+        # TODO: Add paths that should trigger tests for AML
+        - test/new-e2e/tests/agent-metrics-logs/**/*
+  - when: manual
+    allow_failure: true
+
+.on_cws_or_e2e_changes_or_manual:
+  - <<: *if_disable_e2e
+    when: never
+  - <<: *if_auto_e2e
+    when: on_success
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
For jobs that specify needs, we could have `always` for the full pipeline

Reply:
I think `on_success` is fine, it prevents us from running a job while the job that is supposed to deploy the package/image failed before

Reply:
I also think that every job should declare its dependencies. I will add the missing ones